exla/test/support/exla_case.ex

```diff
@@ -0,0 +1,46 @@
+defmodule EXLA.Case do
+  @moduledoc """
+  Test case for tensor assertions
+  """
+
+  use ExUnit.CaseTemplate
+
+  using do
+    quote do
+      import EXLA.Case
+    end
+  end
+
+  defmacro assert_equal(left, right) do
+    # Assert against binary backend tensors to show diff on failure
+    quote do
+      assert unquote(left) |> to_binary_backend() == unquote(right) |> to_binary_backend()
+    end
+  end
```

Comment:
This makes the `code:` (in the error message) a bit obfuscated, but still more detailed than a simple exception. The important part is that we show the diff.